Automattic/woocommerce-payments
# Fix: Open two checkout tabs, and use 3DS cards with non-UPE flow

Fixes #5678

### Changes proposed in this Pull Request

In this PR, I am approaching to cancel the attached payment intent (if it exists) rather than re-use it as originally planned in the PR. 

The main benefit of this approach is that it's simple, and does not need to involve any change in the server. 

**For the approach of re-using the attached payment intent**, it requires to two requests: one to update intent, and another one to [confirm intent](https://stripe.com/docs/api/payment_intents/confirm). To approach this, there are two ways: 

- Send two requests on the client: update, and then confirm it. This requires a new endpoint for confirming intent. 
- Send on request on the client to a new endpoint such as `update_and_confirm` on the server. On the server, it will update, and then confirm the attached intent.

**Anyway, re-using the attached payment intent might make code on both server and client more complicated** as it requires (1) more branching out as the `confirm` and `update` intent input data are not the same, (2) a change on the server, at least one new endpoint. So I'd like to approach the one in this PR. 









### Testing instructions

- Follow up instructions on #5678. 
- Confirm: only one payment intent is successful, the other one is cancelled. 
- Confirm: on the order under WP Admin, a note like this exists: `Payment cancelled, and removed from order`.





*

-------------------


**Post merge**
